src/runtime/entrypoints/main.ts

```diff
@@ -856,7 +857,9 @@ export async function applyPartials(res: Response): Promise<void> {
 
 const originalHook = options.vnode;
 options.vnode = (vnode) => {
-  assetHashingHook(vnode);
+  // avoid hashing assets when asset path prefixed,
+  // the CDN should handle caching
+  if (!hasAssetPrefix()) assetHashingHook(vnode);
```

Comment:
came across this when writing tests ([caching](https://fresh.deno.dev/docs/concepts/static-files#caching)). 

> Fresh also does this automatically for src and srcset attributes in <img> and <source> HTML tags. These will automatically use “locked” paths if Fresh deems it safe to do so. You can always opt out of this behaviour per tag, by adding the data-fresh-disable-lock attribute.

When using a CDN usually this will be handled there as every build will upload the files again with a different url.

Reply:
a future improvement would be to combine both and make caching only opt-in, less magic. However, this is a breaking change for sites that expects caching is already there so I prefer to leave that as a future improvement.